icu4c/source/i18n/formatted_string_builder.h

```diff
@@ -204,46 +218,59 @@ class U_I18N_API FormattedStringBuilder : public UMemory {
     friend class FormattedValueStringBuilderImpl;
 };
 
+static_assert(
+    std::is_pod<FormattedStringBuilder::Field>::value,
+    "Field should be a POD type for efficient initialization");
+
+// Little trick using the C/C++ comma (expression list) operator to add
+// assertions to the constexpr Field constructor.
+// Note: C++14 allows them in the constructor function body.
+#if U_DEBUG
+#define UPRV_EXPRESSION_LIST_ASSERT(x) U_ASSERT(x)
+#else
+#define UPRV_EXPRESSION_LIST_ASSERT(x) void(x)
```

Comment:
It is a constructor that returns void.

https://stackoverflow.com/questions/39279074/what-does-the-void-in-decltypevoid-mean-exactly

An alternative would be to cast my value to void:

```cpp
#define UPRV_EXPRESSION_LIST_ASSERT(x) static_cast<void>(x)
```

The point is to get rid of the syntax error.  Hmm, actually, maybe it would be better if I left out the `x` argument altogether, like this:

```cpp
#define UPRV_EXPRESSION_LIST_ASSERT(x) void()
```

I like that.

And, actually, let's do this in the actual definition of `U_ASSERT`. :)